packages/viewer/src/modules/map/components/cesium/utils/useAddDataSourceLayer.composable.ts

```diff
@@ -59,6 +59,22 @@ export default function useAddDataSourceLayer(
         }
         try {
             dataSource = await loadingDataSource
+
+            // Check again after async operation
+            const currentViewer = toValue(viewer)
```

Comment:
I don't really like the idea of assigning the viewer value to two constants.

If the role is to fetch the current state of the viewer, I would treat this as a variable and use a `let` and assign the value to it, or check directly with `viewer.value`.